From d54b19d754f319cee8321387a63d2cf3881bcdda Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Tue, 25 Nov 2014 16:34:39 -0500
Subject: [fs] vfs: export __inode_permission() to modules

Message-id: <20141125163439.18115.66159.stgit@warthog.procyon.org.uk>
Patchwork-id: 100924
O-Subject: [PATCH 24/47] vfs: export __inode_permission() to modules
Bugzilla: 985875
RH-Acked-by: Steve Best <sbest@redhat.com>

From: Miklos Szeredi <mszeredi@suse.cz>

We need to be able to check inode permissions (but not filesystem implied
permissions) for stackable filesystems.  Expose this interface for overlayfs.

Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
Original-commit: bd5d08569cc379f8366663a61558a9ce17c2e460
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/internal.h b/fs/internal.h
index 5ed11d9..3b41f3e 100644
--- a/fs/internal.h
+++ b/fs/internal.h
@@ -44,7 +44,6 @@ extern void __init chrdev_init(void);
 /*
  * namei.c
  */
-extern int __inode_permission(struct inode *, int);
 extern int user_path_mountpoint_at(int, const char __user *, unsigned int, struct path *);
 extern int vfs_path_lookup(struct dentry *, struct vfsmount *,
       const char *, unsigned int, struct path *);
diff --git a/fs/namei.c b/fs/namei.c
index fff0e6e..215244f 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -401,6 +401,7 @@ int __inode_permission(struct inode *inode, int mask)
 
  return security_inode_permission(inode, mask);
 }
+EXPORT_SYMBOL(__inode_permission);
 
 /**
  * sb_permission - Check superblock-level permissions
diff --git a/include/linux/fs.h b/include/linux/fs.h
index c43e88b..075a721 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -2320,6 +2320,7 @@ extern sector_t bmap(struct inode *, sector_t);
 #endif
 extern int notify_change(struct dentry *, struct iattr *, struct inode **);
 extern int inode_permission(struct inode *, int);
+extern int __inode_permission(struct inode *, int);
 extern int generic_permission(struct inode *, int);
 
 static inline bool execute_ok(struct inode *inode)
-- 
1.7.1